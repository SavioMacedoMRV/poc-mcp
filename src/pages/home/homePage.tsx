import React from 'react'
import {Texto} from 'components/texto'
import * as S from './homePage.styles'
import {useRouteLoaderData, useNavigate, useLoaderData} from 'react-router-dom'
import {rotas} from 'resources/rotas'
import {Empreendimento} from 'types/interfaces'
import {RegistrarEvento} from 'infrastructure/logs'
import {Evento} from 'types/enums'
import {ContainerPage} from 'components/containerPage'

export const HomePage = () => {
  const navigate = useNavigate()
  const {obra} = useRouteLoaderData('root-template') as {obra: Empreendimento}
  const {info} = useLoaderData() as {info: string}
  
  const aoClicarBotao = () => {
    RegistrarEvento({evento: Evento.HomePageClickNavegarOutraPagina})
    navigate(`${rotas.OutraPagina}/12345`)
  }

  const aoPesquisar = (texto?: string | number) => {
    // Implement search logic here
    console.log('Pesquisando:', texto)
  }
  
  return (
    <ContainerPage
      cabecalhoProps={{
        pesquisa: {
          aoPesquisar,
          autoPesquisar: true
        },
        filtroPepSuperior: <div>Lista de Hubs</div>,
        filtroAtividade: <div>Filtros</div>,
        aoClicarVoltar: () => navigate(-1)
      }}
    >
      <S.Content>
        <S.Logo />
        <S.StyledSvg />
        <Texto tamanho={18} estilo="semibold">
          Portal 360 - Hub de Formulários
        </Texto>
        {obra && (
          <>
            <Texto tamanho={18}>Obra do Host: {obra.nome}</Texto>
            <Texto tamanho={18}>Status da API: {info}</Texto>
            <S.BotaoNavegar
              data-testid={'btn-navegar-outra-pagina'}
              texto="Acessar Formulários"
              aoClicar={aoClicarBotao}
            />
          </>
        )}
      </S.Content>
    </ContainerPage>
  )
}
