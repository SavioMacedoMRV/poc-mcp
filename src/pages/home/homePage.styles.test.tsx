import {render} from '@testing-library/react'
import * as S from './homePage.styles'

describe('HomePage.styles', () => {
  test('BotaoNavegar DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.BotaoNavegar aoClicar={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  test('Container DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Container />)
    expect(container).toMatchSnapshot()
  })

  test('Contente DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Content />)
    expect(container).toMatchSnapshot()
  })

  test('Logo DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Logo />)
    expect(container).toMatchSnapshot()
  })

  test('StyledSvgDEVE ser igual ao snapshot', () => {
    const {container} = render(<S.StyledSvg />)
    expect(container).toMatchSnapshot()
  })
})
