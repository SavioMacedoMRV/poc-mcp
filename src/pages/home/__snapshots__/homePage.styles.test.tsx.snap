// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage.styles BotaoNavegar DEVE ser igual ao snapshot 1`] = `
.c1 {
  font-weight: 400;
  font-size: 12px;
  line-height: 24px;
  font-family: AvertaStd-Semibold;
  color: #FFFFFF;
  text-align: start;
  cursor: pointer;
  text-transform: none;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: auto;
  height: auto;
  cursor: pointer;
  opacity: 1;
  background: #54BC82;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

<div>
  <button
    class="c0 "
    cor="#54BC82"
  >
    <span
      alturalinha="24"
      class="c1 "
      cor="#FFFFFF"
      cursor="pointer"
      data-testid="texto-botao"
      estilo="semibold"
    />
    <svg
      data-file-name="SvgISeta"
      fill="#FFFFFF"
      height="24"
      style="min-width: 24px; min-height: 24px; max-width: 24px; max-height: 24px; transform: rotate(0deg);"
      width="24"
    />
  </button>
</div>
`;

exports[`HomePage.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  background: #EDF2F0;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`HomePage.styles Contente DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding: 20px;
  height: 100vh;
  gap: 16px;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`HomePage.styles Logo DEVE ser igual ao snapshot 1`] = `
.c0 {
  height: 21px;
  width: 125px;
}

<div>
  <svg
    class="c0"
    data-file-name="SvgILogoMRV"
  />
</div>
`;

exports[`HomePage.styles StyledSvgDEVE ser igual ao snapshot 1`] = `
.c0 {
  fill: red;
  width: 32px;
  height: 32px;
}

<div>
  <svg
    class="c0"
    data-file-name="SvgIUser"
  />
</div>
`;
